cmd/hamctl/command/root.go

```diff
@@ -58,8 +58,9 @@ func NewRoot(version *string) (*cobra.Command, error) {
 				committer, err := gitConfigAPI.CommitterDetails()
 				if err != nil {
 					missingFlags = append(missingFlags, "user-email")
+				} else {
```

Comment:
This really feels wrong. If `err != nil` I would expect `comitter` to at least be defined. Always.
Isn't this the API of `gitConfigAPI` that is not behaving well?

Reply:
I don't really agree. Right now in the code, we don't exit even if we have an error (until further down the line), however, we still depend on the committer struct being present, which in this case it isn't because it is a reference. We could change it to a struct, but it feels wrong to depend on default values on an object that shouldn't be set (if an error occurs).

I.e. if err is defined we only trust the error value and nothing else. 
If the err is not defined we can trust committer, because that is the api.

Also the only reason why we need the else, is because something further down are bundling up these missing flags and throwing an error.

Reply:
I see 👍 Makes fine sense. I misread the `if` condition here and expected "it handles the error case" where in fact it just collects flag error cases and not flow control 🙈